lib/class-oa2-authenticator.php

```diff
@@ -34,14 +34,13 @@ public function __construct() {
    * @return WP_User|null|WP_Error Authenticated user on success, null if no OAuth data supplied, error otherwise
    */
   public function authenticate( $user ) {
-	if ( !empty( $user ) ) {
-	  return $user;
-	}
-
 	// Check if token is in request
 	$bearer_token = OA2_Header_Helper::get_authorization_bearer();
 	if ( empty( $bearer_token ) ) {
 	  $this->auth_status = $bearer_token;
+      if ( !empty( $user ) ) {
```

Comment:
I actually reviewed my own code through again and making the change makes sense. It is more unlikely that the user is sending a Bearer token with the request (you don't do that by accident) but you might send cookies unintentionally (because they are sent automatically), if you're e.g. using an app in the browser. We just need to add a check to the beginning where we make sure that we don't run the authentication checks every time `determine_current_user` is run (which is quite often). I'll make that change and then merge these fixes.